lib/ansible/modules/cloud/azure/azure_rm_batchaccount.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/python
+#
+# Copyright (C) 2019 Junyi Yi (@JunyiYi)
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_batchaccount
+version_added: "2.9"
+
+short_description: Manages a Batch Account on Azure.
+
+description:
+    - Create, update and delete instance of Azure Batch Account.
+
+options:
+    resource_group:
+        description:
+        - The name of the resource group in which to create the Batch Account.
+        required: true
+    name:
+        description:
+        - The name of the Batch Account.
+        required: true
+    location:
+        description:
+        - Specifies the supported Azure location where the resource exists.
+        required: true
+    auto_storage_account_id:
```

Comment:
this option may be actually renamed to "auto_storage_account" and should accept both string that could be an id, or could be just storage account name.
if user specifies just name, current resource group should be assumed by default.
 or dict containing "resource_group" / "storage_account_name", or just "storage_account_name".